src/parachain/escrow.ts

```diff
@@ -210,29 +210,38 @@ export class DefaultEscrowAPI implements EscrowAPI {
         amount: MonetaryAmount<GovernanceCurrency>;
         apy: Big;
     }> {
-        const [userStake, totalStake, blockReward, stakedBalance, currentBlockNumber, minimumBlockPeriod, maxPeriod] =
-            await Promise.all([
-                this.getEscrowStake(accountId),
-                this.getEscrowTotalStake(),
-                this.getRewardPerBlock(),
-                this.getStakedBalance(accountId),
-                this.systemAPI.getCurrentBlockNumber(),
-                this.api.consts.timestamp.minimumPeriod,
-                this.getMaxPeriod(),
-            ]);
-
-        return estimateReward(
-            this.governanceCurrency,
-            userStake,
-            totalStake,
-            blockReward,
-            stakedBalance,
-            currentBlockNumber,
-            minimumBlockPeriod.toNumber(),
-            maxPeriod.toNumber(),
-            amountToLock,
-            blockLockTimeExtension
+        const stakedBalance = await this.getStakedBalance(accountId);
+
+        let atomicStakedAmount: Big = stakedBalance.amount.toBig(ATOMIC_UNIT);
+
+        let nullableAmountToCheck: bigint | null = null;
+        if (amountToLock && !amountToLock.isZero()) {
+            atomicStakedAmount = atomicStakedAmount.add(amountToLock.toBig(ATOMIC_UNIT));
+            // get as bigint for the rpc api
+            nullableAmountToCheck = BigInt(atomicStakedAmount.toString());
+        }
+
+        let nullableBlockToEnd: bigint | null = null;
+        if (blockLockTimeExtension) {
+            const currentEndBlock = !stakedBalance.amount.isZero() ? BigInt(stakedBalance.endBlock) : BigInt(0);
+            nullableBlockToEnd = currentEndBlock + BigInt(blockLockTimeExtension);
+        }
+
+        const rawRewardRate = await this.api.rpc.reward.estimateEscrowRewardRate(
+            accountId,
+            nullableAmountToCheck,
+            nullableBlockToEnd
```

Comment:
Okay so probably I should've been more clear about this before (I'll add a comment to the RPC later) but actually these are the amount & length to extend. It already considers the staked balance and end lock time, so you can make the above function even simpler!

Reply:
> Okay so probably I should've been more clear about this before (I'll add a comment to the RPC later) but actually these are the amount & length to extend. It already considers the staked balance and end lock time, so you can make the above function even simpler!

That makes me unreasonable happy. On it.